daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1129,28 +1131,36 @@ private[lf] object SBuiltin {
                 IE.ContractNotActive(coid, expectedTemplateIdOpt.getOrElse(ifaceId), nid)
               )
             )
-          checkTemplateId(cached.templateId) {
-            machine.returnValue = cached.value
-          }
+          if (
+            machine.compiledPackages
+              .getDefinition(ImplementsDefRef(cached.templateId, ifaceId))
+              .isEmpty
+          )
```

Comment:
This check doesn't work if the template package isn't loaded.

Reply:
The contract is cached, so the package is already loaded.

Reply:
Oh sorry, I missed that